### ISSUE TITLE
User Reset Password Not Redirected to Confirmation Page

### ISSUE THREAD

**Issue by jeffolmstead** [OrchardCMS/OrchardCore#3428]
Related to #3426 is that when a user has filled in their Email or user name to reset their password, they are not redirected to the confirmation screen letting them know that an email has been sent to them. The code in OrchardCore.Users.Controllers.ResetPasswordController --> ForgotPassword ends like this:
```
var resetPasswordUrl = Url.Action("ResetPassword", "ResetPassword", new { code = user.ResetToken }, HttpContext.Request.Scheme);
// send email with callback link
await SendEmailAsync(user.Email, T["Reset your password"], new LostPasswordViewModel() { User = user, LostPasswordUrl = resetPasswordUrl });

await _passwordRecoveryFormEvents.InvokeAsync(i => i.PasswordRecoveredAsync(), _logger);
```
And I believe it should add one additional line at the end to redirect the user to the confirmation screen:
```
var resetPasswordUrl = Url.Action("ResetPassword", "ResetPassword", new { code = user.ResetToken }, HttpContext.Request.Scheme);
// send email with callback link
await SendEmailAsync(user.Email, T["Reset your password"], new LostPasswordViewModel() { User = user, LostPasswordUrl = resetPasswordUrl });

await _passwordRecoveryFormEvents.InvokeAsync(i => i.PasswordRecoveredAsync(), _logger);

return RedirectToLocal(Url.Action("ForgotPasswordConfirmation", "ResetPassword"));
```
For something more sophisticated, could check to see if "SendEmailAsync" returned "true" and notify if there was some reason the email did not send... We don't do that on other items though (e.g. SendEmailConfirmationTokenAsync doesn't check) so perhaps above is good enough.

**Comment by sebastienros** [comment]
My recommendation is to look at what the ASP.NET templates do, as it has originated from these.

**Comment by Skrypt** [comment]
I agree with him ; I needed to add this last line to my controller to make it work.

**Comment by sebastienros** [comment]
I am not disagreeing with the issue, but will just blindly accept any PR that reflects the ASP.NET templates code, just for security reasons. If you provide a PR that fixed the issue and reflects it, it's merged in 2.34 seconds

**Comment by jeffolmstead** [comment]
On ASP.NET Identity samples [AccountController](https://github.com/aspnet/AspNetCore/blob/master/src/Identity/samples/IdentitySample.Mvc/Controllers/AccountController.cs) on line 283 (commented out) they would return the View for ForgotPasswordConfirmation, equivalent to what I have done above. I will prepare the pull request.
